runner/src/scenario/ceramic/new_streams.rs

```diff
@@ -207,20 +237,24 @@ async fn instantiate_small_model(
     user: &mut GooseUser,
     store_in_redis: bool,
     conn: Arc<tokio::sync::Mutex<MultiplexedConnection>>,
+    only_once_per_network: bool,
 ) -> TransactionResult {
-    let user_data = CeramicModelInstanceTestUser::user_data(user).to_owned();
-    let response = ModelInstanceRequests::create_model_instance(
-        user,
-        user_data.user_cli(),
-        &user_data.small_model_id,
-        "instantiate_small_model_instance",
-        &models::SmallModel::random(),
-    )
-    .await?;
-    if store_in_redis {
-        let mut conn: tokio::sync::MutexGuard<'_, MultiplexedConnection> = conn.lock().await;
-        let stream_id_string = response.to_string();
-        let _: () = conn.sadd("anchor_mids", stream_id_string).await.unwrap();
+    let should_create = !only_once_per_network || is_goose_lead_worker();
+    if should_create {
+        let user_data = CeramicModelInstanceTestUser::user_data(user).to_owned();
+        let response = ModelInstanceRequests::create_model_instance(
+            user,
+            user_data.user_cli(),
+            &user_data.small_model_id,
+            "instantiate_small_model_instance",
+            &models::SmallModel::random(),
+        )
+        .await?;
+        if store_in_redis {
+            let mut conn: tokio::sync::MutexGuard<'_, MultiplexedConnection> = conn.lock().await;
+            let stream_id_string = response.to_string();
+            let _: () = conn.sadd("anchor_mids", stream_id_string).await.unwrap();
```

Comment:
continuing my comment from above, if the connection is dead, we could grab `user_data.redis_cli()` and make/store a new one